rasa_nlu/evaluate.py

```diff
@@ -616,39 +639,70 @@ def remove_duckling_entities(entity_predictions):
     return patched_entity_predictions
 
 
-def run_evaluation(data_path, model_path,
+def run_evaluation(data_path, model,
                    errors_filename='errors.json',
                    confmat_filename=None,
                    intent_hist_filename=None,
                    component_builder=None):  # pragma: no cover
     """Evaluate intent classification and entity extraction."""
 
     # get the metadata config from the package data
-    interpreter = Interpreter.load(model_path, component_builder)
+    if isinstance(model, Interpreter):
+        interpreter = model
+    else:
+        interpreter = Interpreter.load(model, component_builder)
     test_data = training_data.load_data(data_path,
                                         interpreter.model_metadata.language)
     extractors = get_entity_extractors(interpreter)
     entity_predictions, tokens = get_entity_predictions(interpreter,
                                                         test_data)
+
     if duckling_extractors.intersection(extractors):
         entity_predictions = remove_duckling_entities(entity_predictions)
         extractors = remove_duckling_extractors(extractors)
 
+    result = {
+        "intent_evaluation": None,
+        "entity_evaluation": None
+    }
+
     if is_intent_classifier_present(interpreter):
         intent_targets = get_intent_targets(test_data)
         intent_results = get_intent_predictions(
                 intent_targets, interpreter, test_data)
         logger.info("Intent evaluation results:")
 
-        evaluate_intents(intent_results, errors_filename, confmat_filename,
-                         intent_hist_filename)
+        report, precision, f1, accuracy = evaluate_intents(intent_results,
+                                                           errors_filename,
+                                                           confmat_filename,
+                                                           intent_hist_filename)
+
+        predictions = [
+            {
+                "text": res.message,
+                "intent": res.target,
+                "predicted": res.prediction,
+                "confidence": res.confidence
+            } for res in intent_results
+        ]
+
+        result['intent_evaluation'] = {
+            "predictions": predictions,
+            "report": report,
+            "precision": precision,
+            "f1_score": f1,
+            "accuracy": accuracy
+        }
 
     if extractors:
         entity_targets = get_entity_targets(test_data)
 
         logger.info("Entity evaluation results:")
-        evaluate_entities(entity_targets, entity_predictions, tokens,
-                          extractors)
+        result['entity_evaluation'] = evaluate_entities(entity_targets,
```

Comment:
i like this structure (call to function that returns the entity specific result) - can we do the same for intent? (also to shorten this function)